projects/vdk-core/src/vdk/internal/builtin_plugins/job_properties/datajobs_service_properties.py

```diff
@@ -63,8 +69,29 @@ def get_all_properties(self) -> Dict[str, PropertyValue]:
 
     def set_all_properties(self, properties: Dict[str, PropertyValue]) -> None:
         """
-        Persists the passed properties overwriting all previous properties
+        Invokes the write pre-processors if any such configured.
+        Persists the passed properties overwriting all previous properties.
         """
+        if self._write_preprocessors:
+            properties = deepcopy(
+                properties
+            )  # keeps the outer scope originally-referenced dictionary intact
+            for client in self._write_preprocessors:
+                try:
+                    properties = client.write_properties(
+                        self._job_name, self._team_name, properties
+                    )
+                except Exception as e:
+                    log_and_throw(
+                        to_be_fixed_by=ResolvableBy.USER_ERROR,
```

Comment:
Are we sure this would always be a `User Error`? If something happens due to an issue with a plugin or a backend, this may not be in user's control.

Reply:
Properties write preprocessors are designed for user-specific customisations. I believe the case is the same with ingestion preprocessors, where error-handling approach matches

Reply:
Ok.